### DISCUSSION TITLE
Introducing Staking and Slash to establish ORMP Oracle SLA

### DISCUSSION THREAD

**Discussion by hackfisher** [ringecosystem/discussion#2] [General]
Supported Chains: The Darwinia Msgport supports various chains as listed in [Supported Chains](https://github.com/darwinia-network/darwinia-msgport/blob/main/SUPPORTED.md).

Oracle Provision: The SubAPI provides an oracle for the message root, utilizing version 2 of the ORMP oracle as detailed on [ORMP Oracle v2](https://github.com/darwinia-network/ormpipe).

SubAPIMultiSig Service Level Challenges and Enhancements:

Engineering Bugs: There could be existing bugs within the engineering of the oracle systems.

SLA and Incentive for Oracle DevOps: It's proposed to increase the fees paid to SubAPI by 20% to enhance the Service Level through the use of SubAPI Oracle.

Staking and Slash Mechanism for SLA Failure: If the Service Level Agreement (SLA) is not met, a Slash mechanism requiring Staking will be established to penalize failures.

Oracle Relayer Engineering and DevOps: To address single points of failure, the proposal includes establishing redundant backup mechanisms and solutions for Oracle Relayer engineering bugs and DevOps issues.

Regular Settlements for SubAPI: Settlement rules will be established for SubAPI with a settlement period of every two months.
Staking RING for Bidding: Bidding will be based on Staking RING, selecting the top five bids in descending order, subject to audit, with a minimum of 25,000 RING. This will be updated every two months.

Penalties for Failures: In the event of a fault, unsigned Oracle Signers will not share in fee revenue for that settlement period and will be slashed 5% of their Staking for failures.

Appeals for Software Design Faults: Appeals for faults due to software design, rather than operational issues, can be made to the SubAPIDAO for arbitration.

**Comment by jiguantong** [comment]
https://github.com/darwinia-network/ORMP/pull/121

Oracle Income
Relayer: 84% (Cost)
SubAPIDao: 16% (Benefit)

**Comment by jiguantong** [comment]
## SLA
One period: 2 months
If the uptime percentage is lower than 95%, the revenue in that period will be lost

| One period Uptime Percentage | Percentage of staking will be slashed |
|:---:|:---:|
| 98.00% - < 99.00% | 5% |
| 95.00% - < 98.00% | 10% |
| < 95.00% | 50%|

**Comment by hackfisher** [comment]
Instead of increasing price which may not good for current stage of go-to-market, this is an alternative solution by proposal to Darwinia Treasury to allocate incentive for this subapi node staking.

https://github.com/orgs/darwinia-network/discussions/1423
